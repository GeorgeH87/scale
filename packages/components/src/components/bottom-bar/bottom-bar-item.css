:host {
  --selected-color: var(--telekom-color-text-and-icon-primary-standard);
  --label-size: var(--telekom-typography-font-size-small);
  --label-weight: var(--telekom-typography-font-weight-medium);
}

.bottom-bar-item {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex: 1;
  height: 100%;
}

.bottom-bar-item--selected {
  color: var(--selected-color);
}

.bottom-bar-item__label {
  font-size: var(--label-size);
  font-weight: var(--label-weight);
  margin-top: 4px;
}

:host-context([data-platform='android']) .scale-bottom-bar-item__container {
  background-color: var(--telekom-color-ui-subtle);
  border-radius: var(--telekom-radius-pill);
  padding: 7px 22px 7px 22px;
  display: flex;
}
