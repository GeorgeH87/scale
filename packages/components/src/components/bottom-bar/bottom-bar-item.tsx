/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

 import { Component, Prop, h, Host } from '@stencil/core';
 import classNames from 'classnames';
import { renderIcon } from '../../utils/render-icon';
 @Component({
   tag: 'scale-bottom-bar-item',
   styleUrl: 'bottom-bar-item.css',
   shadow: true,
 })
 export class Card {
   /** (optional) Icon label */
   @Prop() icon?: string;      
   /** (optional) Icon label */
   @Prop() label?: string;
   /** (optional) Selected state */
   @Prop() selected: boolean = false;     
   /** (optional) Injected CSS styles */
   @Prop() styles?: string;
 
   render() {
     return (
       <Host>
         {this.styles && <style>{this.styles}</style>}
           <div
             class={this.getCssClassMap()}
            //  part={classNames('base', !!this.to && 'interactive')}
           >
              <slot name="icon"/>
                    {this.icon && renderIcon({
                    tag: `scale-icon-${this.icon}`,
                    attributes: {
                        size:"18",
                        selected: this.selected,
                        color: this.selected ? `var(--selected-color)` : `var(--telekom-color-text-and-icon-standard)`
                    }})
                }
              {this.label && <span class="bottom-bar-item__label"> {this.label} </span>}
               
           </div>
       </Host>
     );
   }
 
   getCssClassMap() {
     return classNames('bottom-bar-item', 
     this.selected && 'bottom-bar-item--selected');
   }
 }
 