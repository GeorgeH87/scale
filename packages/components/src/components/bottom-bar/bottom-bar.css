:host {
}

.bottom-bar {
  display: flex;
  justify-content: space-between;
  height: 48px;
  padding-left: 36px;
  padding-right: 36px;
  box-shadow: var(--telekom-shadow-app-bar-bottom-flat);
}

::slotted(scale-bottom-bar-item) {
  flex: 1;
}

:host-context([data-platform='android']) .bottom-bar {
  height: 80px;
}

:host-context([data-platform='ios']) .bottom-bar {
  height: 48px;
}
