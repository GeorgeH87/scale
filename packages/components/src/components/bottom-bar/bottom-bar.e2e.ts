/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newE2EPage } from '@stencil/core/testing';

describe('scale-bottom-bar', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent(`
        <scale-bottom-bar>
            <scale-bottom-bar-item icon="home-home" selected></scale-bottom-bar-item>
            <scale-bottom-bar-item icon="action-favorite"></scale-bottom-bar-item>
            <scale-bottom-bar-item icon="content-music"></scale-bottom-bar-item>
            <scale-bottom-bar-item icon="user-file-user"></scale-bottom-bar-item>  
        </scale-breadcrumb>    
    `);
    const element = await page.find('scale-bottom-bar');
    expect(element).toHaveClass('hydrated');
  });

  it('should add selected attribute to clicked bottom bar item', async () => {
    const page = await newE2EPage();
    await page.setContent(`
        <scale-bottom-bar>
            <scale-bottom-bar-item id="first" icon="home-home" selected></scale-bottom-bar-item>
            <scale-bottom-bar-item id="second" icon="action-favorite"></scale-bottom-bar-item>
            <scale-bottom-bar-item id="third" icon="content-music"></scale-bottom-bar-item>
            <scale-bottom-bar-item id="fourth" icon="user-file-user"></scale-bottom-bar-item>  
        </scale-breadcrumb>    
    `);
    await (await page.find('scale-bottom-bar-item#second')).click();
    const element = await page.find('scale-bottom-bar-item#second');
    expect(element.getAttribute('selected')).toBeTruthy;
  });

  //   it('should accept a custom separator via prop', async () => {
  //     const page = await newE2EPage();
  //     await page.setContent(`
  //       <scale-breadcrumb separator="/">
  //         <a href="/a">A</a>
  //         <a href="/b">B</a>
  //       </scale-breadcrumb>
  //     `);
  //     const element = await page.find('scale-breadcrumb >>> span');
  //     expect(element).toEqualText('/');
  //   });

  //   it('should accept a custom separator via slot', async () => {
  //     const page = await newE2EPage();
  //     await page.setContent(`
  //       <scale-breadcrumb separator="/">
  //         <span slot="separator">
  //           <span>/</span>
  //         </span>
  //         <a href="/a">A</a>
  //         <a href="/b">B</a>
  //       </scale-breadcrumb>
  //     `);
  //     const separator = await page.find('scale-breadcrumb >>> span');
  //     expect(separator.innerHTML.trim()).toBe('<span>/</span>');
  //     const list = await page.findAll('scale-breadcrumb >>> li');
  //     expect(list.length).toBe(2);
  //   });

  //   it('should add aria-current="page" attribute to last item, when link', async () => {
  //     const page = await newE2EPage();
  //     await page.setContent(`
  //       <scale-breadcrumb separator="/">
  //         <a href="/a">A</a>
  //         <a href="/b">B</a>
  //       </scale-breadcrumb>
  //     `);
  //     const element = await page.find('scale-breadcrumb >>> li:last-child a');
  //     expect(element.getAttribute('aria-current')).toBe('page');
  //   });

  //   it('should render last item as plain text when no href is found', async () => {
  //     const page = await newE2EPage();
  //     await page.setContent(`
  //       <scale-breadcrumb separator="/">
  //         <a href="/a">A</a>
  //         <span>B</span>
  //       </scale-breadcrumb>
  //     `);
  //     const element = await page.find('scale-breadcrumb >>> li:last-child span');
  //     expect(element).toEqualText('B');
  //   });
});
