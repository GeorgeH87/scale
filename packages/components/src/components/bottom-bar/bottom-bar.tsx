/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

 import { Component, Prop, h, Host, Listen, Element } from '@stencil/core';
 import classNames from 'classnames';
 @Component({
   tag: 'scale-bottom-bar',
   styleUrl: 'bottom-bar.css',
   shadow: true,
 })
 export class Card {
  @Element() hostElement: HTMLElement;

   /** (optional) Elements label */
   @Prop() elements?: any = []

   /** (optional) Injected CSS styles */
   @Prop() styles?: string;
 
   @Listen('click')
   handleClick(event: MouseEvent) {
    const selected = event.target as HTMLElement
    if (selected.tagName !== 'SCALE-BOTTOM-BAR-ITEM') {
      return 
    }
    const allBottomBarItems = Array.from(this.hostElement.querySelectorAll('scale-bottom-bar-item'));
    allBottomBarItems.forEach(el => {
      el.removeAttribute('selected')
      if (el === event.target) {
        el.setAttribute('selected', '')
      }
    })
   }

   render() {
    console.log('this.elements', this.elements)
     return (
       <Host>
         {this.styles && <style>{this.styles}</style>}
           <div
             class={this.getCssClassMap()}
           >
              <slot />
           </div>
       </Host>
     );
   }
 
   getCssClassMap() {
     return classNames('bottom-bar', 'bottom-bar');
   }
 }
 