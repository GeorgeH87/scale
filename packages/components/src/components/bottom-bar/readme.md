# scale-bottom-bar-item



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                    | Type      | Default     |
| ---------- | ---------- | ------------------------------ | --------- | ----------- |
| `icon`     | `icon`     | (optional) Icon label          | `string`  | `undefined` |
| `label`    | `label`    | (optional) Icon label          | `string`  | `undefined` |
| `selected` | `selected` | (optional) Selected state      | `boolean` | `false`     |
| `styles`   | `styles`   | (optional) Injected CSS styles | `string`  | `undefined` |


## Dependencies

### Depends on

- [scale-icon](../icon)

### Graph
```mermaid
graph TD;
  scale-bottom-bar-item --> scale-icon
  style scale-bottom-bar-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
