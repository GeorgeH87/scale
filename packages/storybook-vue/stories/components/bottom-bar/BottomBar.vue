<template>
  <scale-bottom-bar
    :icon="icon"
    :label="label"
    :styles="styles"
  >
    <slot></slot>
    <slot name="separator"></slot>
  </scale-bottom-bar>
</template>

<script>
export default {
  props: {
    separator: String,
    label: String,
    styles: { type: String }
  },
}
</script>
