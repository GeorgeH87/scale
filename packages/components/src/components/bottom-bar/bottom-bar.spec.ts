import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { BottomBar } from './bottom-bar';



describe('Bottom Bar', () => {
    it('should match snapshot', async () => {
        const page = await newSpecPage({
            components: [BottomBar],
            html: `
            <scale-bottom-bar>
                <scale-bottom-bar-item icon="home-home" selected></scale-bottom-bar-item>
                <scale-bottom-bar-item icon="action-favorite"></scale-bottom-bar-item>
                <scale-bottom-bar-item icon="content-music"></scale-bottom-bar-item>
                <scale-bottom-bar-item icon="user-file-user"></scale-bottom-bar-item>  
            </scale-bottom-bar>
            `
        });        
        expect(page.root).toMatchSnapshot();
    });
})