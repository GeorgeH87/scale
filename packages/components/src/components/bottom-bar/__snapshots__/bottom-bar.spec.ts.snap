// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Bar should match snapshot 1`] = `
<scale-bottom-bar>
  <mock:shadow-root>
    <div class="bottom-bar">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <scale-bottom-bar-item icon="home-home" selected=""></scale-bottom-bar-item>
  <scale-bottom-bar-item icon="action-favorite"></scale-bottom-bar-item>
  <scale-bottom-bar-item icon="content-music"></scale-bottom-bar-item>
  <scale-bottom-bar-item icon="user-file-user"></scale-bottom-bar-item>
</scale-bottom-bar>
`;
